lib/charms/opensearch/v0/opensearch_config.py

```diff
@@ -206,9 +206,12 @@ def remove_temporary_data_role(self):
     def add_seed_hosts(self, cm_ips: List[str]):
         """Add CM nodes ips / host names to the seed host list of this unit."""
         cm_ips_set = set(cm_ips)
-        with open(self._opensearch.paths.seed_hosts, "w+") as f:
-            lines = "\n".join([entry for entry in cm_ips_set if entry.strip()])
-            f.write(f"{lines}\n")
+
+        # only update the file if there is data to update
+        if cm_ips_set:
```

Comment:
when can there ever be no seed_hosts?

Reply:
has been adjusted so that `cm_ips` is always available.